examples/spark-pi-prometheus-servlet.yaml

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright 2024 The Kubeflow authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: sparkoperator.k8s.io/v1beta2
+kind: SparkApplication
+metadata:
+  name: spark-pi-python
+spec:
+  type: Python
+  pythonVersion: "3"
+  mode: cluster
+  image: spark:3.5.3
+  imagePullPolicy: IfNotPresent
+  mainApplicationFile: local:///opt/spark/examples/src/main/python/pi.py
+  sparkVersion: 3.5.3
+  sparkConf:
+    # Expose Spark metrics for Prometheus
```

Comment:
Please validate this upstream config to see if we need to more config

https://github.com/apache/spark/blob/44966c94b686df72b827307d3c3eadc6804abaf2/conf/metrics.properties.template#L116C1-L123C60

https://github.com/apache/spark/blob/44966c94b686df72b827307d3c3eadc6804abaf2/conf/metrics.properties.template#L208C1-L212C75

Reply:
It doesn't seem like you need to them. I added annotations for Prometheus though.